src/main/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImpl.java

```diff
@@ -62,6 +62,7 @@ public void anonymizeCourtCaseEntity(UserAccountEntity userAccount, CourtCaseEnt
         courtCase.getDefenceList().forEach(defenceEntity -> anonymizeDefenceEntity(userAccount, defenceEntity));
         courtCase.getProsecutorList().forEach(prosecutorEntity -> anonymizeProsecutorEntity(userAccount, prosecutorEntity));
         courtCase.getHearings().forEach(hearingEntity -> anonymizeHearingEntity(userAccount, hearingEntity));
+        eventService.getAllCourtCaseEventVersions(courtCase).forEach(eventEntity -> anonymizeEventEntity(userAccount, eventEntity));
```

Comment:
You may need to manually save these events as they may not all get caught up in the saveCase method

Reply:
no need, this service is marked as `@Transactional`. The events have been retrieved with a call to a JPA repo so they are managed by JPA which will flush changes to DB once the method completes